### PR TITLE
[templates] add a new maui-blank template

### PR DESCRIPTION
**Pull request by jonathanpeppers**
Since ceaff6b3 added Shell to the `maui` template, there is a slight
gap if you literally want a MAUI project with nothing in it. This adds
a project with a single `Label` that has the setup for XAML and a
splash screen.

It sets up the `MainPage` such as:

    MainPage = new Shell
    {
        CurrentItem = new MainPage()
    };

This minimal use of Shell enables the navigation APIs so they actually
work -- enabling you to push/pop pages.

This new `maui-blank` template is displayed by `dotnet new -l` such
as:

    Template Name                                 Short Name           Language    Tags
    --------------------------------------------  -------------------  ----------  ---------------------------------------------------
    .NET MAUI App (Preview)                       maui                 [C#]        MAUI/Android/iOS/macOS/Mac Catalyst/Windows
    .NET MAUI Blank App (Preview)                 maui-blank           [C#]        MAUI/Android/iOS/macOS/Mac Catalyst/Windows
    .NET MAUI Blazor App (Preview)                maui-blazor          [C#]        MAUI/Android/iOS/macOS/Mac Catalyst/Windows/Blazor
    .NET MAUI Class Library (Preview)             mauilib              [C#]        MAUI/Android/iOS/macOS/Mac Catalyst/Windows
    .NET MAUI ContentPage (C#) (Preview)          maui-page-csharp     [C#]        MAUI/Android/iOS/macOS/Mac Catalyst/WinUI/Xaml/Code
    .NET MAUI ContentPage (XAML) (Preview)        maui-page-xaml       [C#]        MAUI/Android/iOS/macOS/Mac Catalyst/WinUI/Xaml/Code
    .NET MAUI ContentView (C#) (Preview)          maui-view-csharp     [C#]        MAUI/Android/iOS/macOS/Mac Catalyst/WinUI/Xaml/Code
    .NET MAUI ContentView (XAML) (Preview)        maui-view-xaml       [C#]        MAUI/Android/iOS/macOS/Mac Catalyst/WinUI/Xaml/Code

### CODE REVIEW

**Review by jsuarezruiz — APPROVED** [review]

File: src/Templates/src/templates/maui-blank/.template.config/ide.host.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+    "$schema": "http://json.schemastore.org/vs-2017.3.host",
+    "icon": "ide/icon.ico",
+    "unsupportedHosts": [
+        {
+            "id": "vs",
+            "version": "(,17.1)"
```
Comment (jfversluis) [inline src/Templates/src/templates/maui-blank/.template.config/ide.host.json:7]:
```suggestion
            "version": "(,17.2)"
```

File: src/Templates/src/templates/maui-blank/.template.config/ide.host.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+        }
+    ]
```
Comment (jfversluis) [inline src/Templates/src/templates/maui-blank/.template.config/ide.host.json:9]:
```suggestion
    ],
    "requiredComponents": [ 
        { 
              "id": "Microsoft.VisualStudio.ComponentGroup.Maui.All", 
              "hostId": "vs", 
              "componentType": "SetupComponent" 
        }
   ]
```

Comment (jfversluis) [reply src/Templates/src/templates/maui-blank/.template.config/ide.host.json:9]: This makes sure the .NET MAUI workload is actually installed for this to show up

File: src/Templates/src/templates/maui-blank/.template.config/ide.host.json (hunk 1 above)
Comment (jfversluis) [reply src/Templates/src/templates/maui-blank/.template.config/ide.host.json:7]: Only show the template for VS 17.2 and up

**Review by jfversluis — CHANGES_REQUESTED** [review]
Also see #5483

File: src/Templates/src/templates/maui-blank/.template.config/ide.host.json (hunk 1 above)
Comment (jonathanpeppers) [reply src/Templates/src/templates/maui-blank/.template.config/ide.host.json:7]: This template was based on the `release/6.0.2xx-preview14` branch, so let me check this.

**Comment by jonathanpeppers** [comment]
This should have the same changes as #5483 now.

**Comment by rachelkang** [comment]
I was thinking the same exact Styles (maybe just minus Shell since it's not Shell) as in the current template - it's a lot, but the idea is that no matter what controls get added, we'll have everything automatically be styled for them

And aww why not dotnet bot! I was thinking the blank would be exactly the same as shell, just without shell ?

**Comment by rachelkang** [comment]
maybe we can at least have the mainpage code in comments?

**Comment by PureWeen** [comment]
@jonathanpeppers something to keep in mind is that with .NET MAUI we removed `Color.Default` so a lot of the Colors that are in the Styles file are really meant as a theme baseline.  Ideally we'd be able to build these styles/themes into the maui.controls itself but we're not really at that point yet. 

It's similar to a blank android or winui app that still has an underlying theme set.

**Review by rmarinho — CHANGES_REQUESTED** [review]
We need to add it to this list..

https://github.com/dotnet/maui/blob/main/eng/automation/LocProject.json

```
 {
              "SourceFile":  ".\\src\\Templates\\src\\templates\\maui-blank\\.template.config\\localize\\templatestrings.json",
              "LclFile":  "loc\\{Lang}\\src\\Templates\\src\\templates\\maui-blank\\.template.config\\localize\\templatestrings.json.lcl",
              "CopyOption":  "LangIDOnName",
              "OutputPath":  ".\\src\\Templates\\src\\templates\\maui-blank\\.template.config\\localize\\"
            },
```

**Comment by mattleibow** [comment]
Does this new template really mean blank? Right now, the only difference is a few less controls on the page and the lack of the shell xaml.

Shouldn't we either make it really blank and just do an App.xaml with maybe a `MainPage = new ContentPage()`? If a new xaml page is needed, then we have the templates for that.

**Comment by jonathanpeppers** [comment]
Let's just revisit this later if developers ask for it.

The reason to use Shell instead of `ContentPage`, is so you can push new pages:
```csharp
MainPage = new Shell
{
    CurrentItem = new MainPage()
};
```

**Pull request closed without merging**